morl_baselines/multi_policy/gpi_pd/gpi_pd.py

```diff
@@ -619,12 +624,19 @@ def _reset_priorities(self, w: th.Tensor):
             rewards_s,
             next_obs_s,
             dones_s,
-        ) = self.replay_buffer.get_all_data(to_tensor=True, device=self.device)
-        num_batches = int(np.ceil(obs_s.size(0) / 1000))
+        ) = self.replay_buffer.get_all_data(to_tensor=False)
+        num_batches = int(np.ceil(obs_s.shape[0] / 1000))
         for i in range(num_batches):
             b = i * 1000
-            e = min((i + 1) * 1000, obs_s.size(0))
+            e = min((i + 1) * 1000, obs_s.shape[0])
             obs, actions, rewards, next_obs, dones = obs_s[b:e], actions_s[b:e], rewards_s[b:e], next_obs_s[b:e], dones_s[b:e]
+            obs, actions, rewards, next_obs, dones = (
+                th.tensor(obs).to(self.device),
```

Comment:
Why this change of not getting it as tensor from the buffer?

Reply:
I thought it would use less memory by not moving the entire replay buffer to GPU.

Reply:
That's smart.